awx/main/scheduler/task_manager.py

```diff
@@ -600,16 +626,17 @@ def reap_jobs_from_orphaned_instances(self):
     def calculate_capacity_consumed(self, tasks):
         self.graph = InstanceGroup.objects.capacity_values(tasks=tasks, graph=self.graph)
 
-    def consume_capacity(self, task, instance_group, instance=None):
+    def consume_capacity(self, task, instance_group, instance=None, impact=None):
+        impact = impact if impact else task.task_impact
         logger.debug(
             '{} consumed {} capacity units from {} with prior total of {}'.format(
-                task.log_format, task.task_impact, instance_group, self.graph[instance_group]['consumed_capacity']
+                task.log_format, impact, instance_group, self.graph[instance_group]['consumed_capacity']
             )
         )
-        self.graph[instance_group]['consumed_capacity'] += task.task_impact
+        self.graph[instance_group]['consumed_capacity'] += impact
         for capacity_type in ('control', 'execution'):
             if instance is None or instance.node_type in ('hybrid', capacity_type):
```

Comment:
If you are consuming capacity for controller_node selection, then `capacity_type` should be "control", not whatever type the unified job is.

Otherwise your approach to call consume_capacity twice is valid and matches my expectations.

Reply:
you're right. I was thinking that the `task` capacity type played into this, but it doesn't, because this is instance-oriented.